common/test/utils.py

```diff
@@ -24,3 +24,14 @@ def write(self, _):
         yield
     finally:
         sys.stderr = savestderr
+
+
+@contextmanager
+def skip_signal(signal, receiver, sender, dispatch_uid=None, weak=False):
```

Comment:
I think `weak` should be `True` here to match the default value for `connect` and `disconnect`. Actually, since you're just passing the arguments straight to those functions without modification you could change this to `def skip_signal(signal, **kwargs):` and then call `signal.disconnect(**kwargs)`.